src/libos/src/time/mod.rs

```diff
@@ -149,21 +149,55 @@ pub fn do_clock_getres(clockid: ClockId) -> Result<timespec_t> {
     Ok(res)
 }
 
+const TIMER_ABSTIME: i32 = 0x01;
+
 pub async fn do_nanosleep(req: &timespec_t, rem: Option<&mut timespec_t>) -> Result<()> {
+    do_clock_nanosleep(ClockId::CLOCK_REALTIME, 0, req, rem).await?;
+    Ok(())
+}
+
+pub async fn do_clock_nanosleep(
+    clockid: ClockId,
+    flags: i32,
+    req: &timespec_t,
+    rem: Option<&mut timespec_t>,
+) -> Result<()> {
+    match clockid {
+        ClockId::CLOCK_REALTIME | ClockId::CLOCK_MONOTONIC | ClockId::CLOCK_BOOTTIME => {}
+        _ => {
+            return_errno!(EINVAL, "clock_id was invalid");
+        }
+    }
+
     let waiter = Waiter::new();
-    let mut duration = Duration::new(req.sec as u64, req.nsec as u32);
+    let mut duration = {
+        if flags == TIMER_ABSTIME {
+            let duration_now = vdso_time::clock_gettime(clockid)?;
```

Comment:
Rename the duration_now with now may be better, because it could help others to understand it.

Reply:
Done.